### PR TITLE
Add an axeCheck helper & use it

### PR DESCRIPTION
**Pull request by bkjohnson**
## Description
Closes <ticket>

This enables us to meet one of the platform's [foundational accessibility tests](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/staging-review-processes.md#foundational-accessibility-tests-required):
- Axe scans included in end-to-end (e2e) tests

This `axeCheck` function will fail on violations of _all_ impacts (`minor`, `moderate`, `serious`, and `critical`). It does exclude some rules by default since they are more appropriate when considering an entire page as opposed to individual components:

- [`document-title`](https://dequeuniversity.com/rules/axe/4.1/document-title)
- [`html-has-lang`](https://dequeuniversity.com/rules/axe/4.1/html-has-lang)
- [`page-has-heading-one`](https://dequeuniversity.com/rules/axe/4.1/page-has-heading-one)
- [`landmark-one-main`](https://dequeuniversity.com/rules/axe/4.1/landmark-one-main)
- [`region`](https://dequeuniversity.com/rules/axe/4.1/region)

## Testing done

- e2e testing

### Failing test

image

### All tests pass

image

## Acceptance criteria
- [ ] Axe scans included in end-to-end (e2e) tests

## Definition of done
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)

### CODE REVIEW

File: src/testing/test-helpers.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import { AxePuppeteer } from '@axe-core/puppeteer';
+
+const ignoredDefaults = ['document-title', 'html-has-lang', 'page-has-heading-one', 'landmark-one-main', 'region'];
```
Comment (bkjohnson) [inline src/testing/test-helpers.ts:3]: These ignored rules don't make much sense when we are looking at an individual component. They seem more useful with an entire page in mind.

File: src/testing/test-helpers.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+const buildAxeErrorOutput = violations =>
+  violations
+    .map(
+      v =>
+        `[${v.impact}] ${v.help}
+Id: ${v.id}
+See ${v.helpUrl}
+${v.nodes.reduce((str, { html, target }) => [str, html, ...target].join('\n\n'), '')}`,
+    )
```
Comment (bkjohnson) [inline src/testing/test-helpers.ts:13]: This output formatting was taken from the existing `axeCheck` function:

https://github.com/department-of-veterans-affairs/component-library/blob/a063396523f9484e730a173f7b520064a3480db2/src/helpers/test-helpers.js#L48-L57

File: src/testing/test-helpers.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    .join('\n') || null;
+
+export async function axeCheck(page: any, additionalIgnored: string[] = []) {
+  const ignoredRules = [...ignoredDefaults, ...additionalIgnored];
+  const results = await new AxePuppeteer(page).disableRules(ignoredRules).analyze();
+
+  const axeError = buildAxeErrorOutput(results.violations);
+  expect(axeError).toBe(null);
```
Comment (bkjohnson) [inline src/testing/test-helpers.ts:21]: I went with allowing `moderate` and `minor` impact violations to cause errors in addition to `serious` and `critical`.

**Comment by bkjohnson** [comment]
Tagging @1Copenut for awareness.

File: src/testing/test-helpers.ts (hunk 1 above)
Comment (cvalarida) [reply src/testing/test-helpers.ts:3]: 1. This sounds good :+1: 
2. This line is super long; we should probably merge / cherry-pick the prettier / eslint changes I made in `master` into this branch :wink:

File: src/testing/test-helpers.ts (hunk 3 above)
Comment (cvalarida) [reply src/testing/test-helpers.ts:21]: This seems fine to me for now. If it becomes a problem later, we can reassess.

**Review by cvalarida — APPROVED** [review]
Looks great!

File: src/testing/test-helpers.ts (hunk 1 above)
Comment (bkjohnson) [reply src/testing/test-helpers.ts:3]: Those changes are merged in, but nothing happens when I save the file. Perhaps we are missing a config to limit line length?

File: src/testing/test-helpers.ts (hunk 3 above)
Comment (1Copenut) [reply src/testing/test-helpers.ts]: Good call! Even `moderate` and `minor` can cause CI breaks upstream, so better to catch them in components.

**Review by 1Copenut — COMMENTED** [review]
👍 Glad to see axe-checks going into component e2e tests.

**Pull request merged**